ister_gui.py

```diff
@@ -350,8 +376,20 @@ def search_swap(choices, config, mount_d):
                 output = output[idx + len(pttr):]
                 output = output[:output.index('"')]
                 if output == 'swap':
-                    disk = ''.join(x for x in part if not x.isdigit())
+                    # first try to set using lsblk -no pkname part
+                    disk = os.path.join('/dev', get_part_devname(part))
+                    # if that fails, fallback to old (unreliable) method
+                    disk = disk or ''.join(x for x in part if not x.isdigit())
                     part = part[len(disk):]
+                    # must be an int
+                    try:
+                        # strip out prefix or other indicators
+                        part = int(''.join(x for x in part if x.isdigit()))
+                    except ValueError:
+                        Alert('Partition error',
+                              'Unable to detect partition number for {}'
+                              .format(part)).do_alert()
```

Comment:
Would it be reasonable to print out the partition type and disk as well do you think? Could assist in error reports.